xdsl/utils/parse_pipeline.py

```diff
@@ -0,0 +1,100 @@
+import re
+from dataclasses import dataclass
+from typing import Iterator
+from xdsl.utils.lexer import Input, Span
+from enum import Enum, auto
+
+
+class Kind(Enum):
+    EOF = auto()
```

Comment:
nit: If you want to be sure no one is using the values you get from `auto`, you can use `object()` instead.